src/lib/operators/join_hash.cpp

```diff
@@ -111,6 +114,9 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
   const unsigned int _partitioning_seed = 13;
   const size_t _radix_bits = 9;
 
+  // Determine correct type for hashing
+  using HashType = typename JoinHashTraits<LeftType, RightType>::HashType;
```

Comment:
Maybe `HashedType` so that people don't think that it's a type of hash?

Reply:
The comparison of different data types in the table scan is probably related to that, am I right? There we just let the compiler decide how to convert the different types. If we decide that implicit conversions make sense, we could add that to the HashTraits (or conversion traits) like this: `using HashType = decltype(R{} + L{})`